types/ramda/index.d.ts

```diff
@@ -893,8 +893,11 @@ export function identity<T>(a: T): T;
  * of the condition predicate.
  */
 // tslint:disable:max-line-length
+export function ifElse<T, TFiltered extends T, TOnTrueResult, TOnFalseResult>(pred: (a: T) => a is TFiltered, onTrue: (a: TFiltered) => TOnTrueResult, onFalse: (a: T) => TOnFalseResult): (a: T) => TOnTrueResult | TOnFalseResult;
+export function ifElse<T, TFiltered extends T, TOnTrueResult, TOnFalseResult>(pred: (a: T) => a is TFiltered, onTrue: (a: TFiltered) => TOnTrueResult, onFalse: (a: T) => TOnFalseResult, a: T): TOnTrueResult | TOnFalseResult;
 export function ifElse<TArgs extends any[], TOnTrueResult, TOnFalseResult>(fn: (...args: TArgs) => boolean, onTrue: (...args: TArgs) => TOnTrueResult, onFalse: (...args: TArgs) => TOnFalseResult): (...args: TArgs) => TOnTrueResult | TOnFalseResult;
-// tslint:enable:max-line-length
+export function ifElse<TArgs extends any[], TOnTrueResult, TOnFalseResult>(fn: (...args: TArgs) => boolean, onTrue: (...args: TArgs) => TOnTrueResult, onFalse: (...args: TArgs) => TOnFalseResult, ...args: TArgs): TOnTrueResult | TOnFalseResult;
+/// tslint:enable:max-line-length
```

Comment:
According to the [source code of `R.ifElse`](https://github.com/ramda/ramda/blob/v0.28.0/source/ifElse.js#L30), it will receive `pred`, `onTrue`, `onFalse` first, and then the datas to process. It can not receive the first three arguments and the `...rest` arguments all at once.

So the first and last signature is incorrect.

Reply:
After using prettier to automatically format code, I think we should remove all `tslint:disable:max-line-length` and `// tslint:enable:max-line-length` gradually.

Reply:
I also saw that and was a bit confused because that's a `tslint` comment and not a `prettier` comment. I guess someone once wanted those definitions to be one-liners so that still comes down to the question of wether that is a good idea or not.